sklearn/compose/_column_transformer.py

```diff
@@ -854,6 +854,14 @@ def _hstack(self, Xs):
                 pd = check_pandas_support("transform")
                 output = pd.concat(Xs, axis=1)
 
+                output_samples = output.shape[0]
+                if any(X.shape[0] != output_samples for X in Xs):
```

Comment:
should we be using `utils.validaiton._num_samples` instead? I don't think these `X`s are guaranteed to have a `shape`, are they?

Reply:
Yea, we can not guarantee it here. I switch it over to using `_num_samples`.